datashader/core.py

```diff
@@ -368,10 +368,25 @@ def bypixel(source, canvas, glyph, agg):
     glyph : Glyph
     agg : Reduction
     """
-    if isinstance(source, pd.DataFrame):
-        dshape = dshape_from_pandas(source)
-    elif isinstance(source, dd.DataFrame):
-        dshape = dshape_from_dask(source)
+    # Avoid datashape.Categorical instantiation bottleneck
+    # by only retaining the necessary columns:
+    # https://github.com/bokeh/datashader/issues/396
+    if 'category' in str(source.dtypes.values):
```

Comment:
Is converting source.dtypes.values to a str the source of the 10% slowdown in the non-categorical case?  If so, is it faster not to convert it to a string?